source/common/http/header_utility.h

```diff
@@ -167,7 +167,7 @@ class HeaderUtility {
   /**
    * @brief a helper function to determine if the headers represent a CONNECT-UDP request.
    */
-  static bool isConnectUdp(const RequestHeaderMap& headers);
+  static bool isConnectUdp(const RequestOrResponseHeaderMap& headers);
```

Comment:
Instead of merging them, can we instead have two methods `isConnectUdpRequest()` and `isConnectUdpResponse()`? That's less likely to cause security issues due to confusion